kiwi/expression.h

```diff
@@ -20,9 +20,13 @@ class Expression
 
     Expression(const Term &term, double constant = 0.0) : m_terms(1, term), m_constant(constant) {}
 
-    Expression(const std::vector<Term> &terms, double constant = 0.0) : m_terms(terms), m_constant(constant) {}
+    Expression(std::vector<Term> terms, double constant = 0.0) : m_terms(std::move(terms)), m_constant(constant) {}
 
-    ~Expression() {}
+    Expression(const Expression&) = default;
+
+    Expression(Expression&&) noexcept = default;
```

Comment:
Trying to build wheels for kiwi on CentOS that is limited to a c++11 compiler I get the following error:
```
./kiwi/expression.h:53:17: error: function ‘kiwi::Expression& kiwi::Expression::operator=(kiwi::Expression&&)’ defaulted on its first declaration with an exception-specification that differs from the implicit declaration ‘kiwi::Expression& kiwi::Expression::operator=(kiwi::Expression&&)’
       Expression& operator=(Expression&&) noexcept = default;
```
Could this be by mistake c++14 only ? or is it a bug in an old gcc version ?

Reply:
ping @SirLynix @Qix-

Reply:
It looks like a bug from an old GCC version, since I don't understand why it wouldn't be noexcept by default here.

Try removing the `noexcept` qualifier for both move constructor / assignment operator.

Reply:
It is very odd, removing this in expression fixes the issue, even though the same pattern exists in many other files. So maybe a weird corner case in the compiler indeed...

Reply:
which GCC version?

Reply:
I think 4.8 but I would need look more at the docker image to be sure.